### ISSUE TITLE
Fix all fortitude C131 issues

### ISSUE THREAD

**Issue by awvwgk** [toml-f/toml-f#200]
add private in module scope

<details>

````
doc/how-to/error/multi/src/dependency.f90:2:1: C131 module 'demo_dependency' missing default accessibility statement
  |
1 | !> Demo module for fpm-like dependency reading
2 | module demo_dependency
  | ^^^^^^^^^^^^^^^^^^^^^^ C131
3 |   use tomlf, only : toml_table, toml_context, toml_key, get_value
4 |   implicit none
  |

doc/how-to/error/single/src/config.f90:2:1: C131 module 'demo_config' missing default accessibility statement
  |
1 | !> Module for reading in configuration data
2 | module demo_config
  | ^^^^^^^^^^^^^^^^^^ C131
3 |   use tomlf, only : toml_table, toml_context, toml_level, get_value
4 |   implicit none
  |

doc/tutorial/fpm-lint/src/config.f90:2:1: C131 module 'fpm_lint_config' missing default accessibility statement
  |
1 | !> Configuration data for the manifest linting
2 | module fpm_lint_config
  | ^^^^^^^^^^^^^^^^^^^^^^ C131
3 |   use tomlf, only : toml_table, toml_context, toml_terminal, toml_error, &
4 |     & toml_stat, get_value
5 |   implicit none
  |

doc/tutorial/json/test/test_lexer.f90:14:1: C131 module 'test_lexer' missing default accessibility statement
   |
12 | ! limitations under the License.
13 |
14 | module test_lexer
   | ^^^^^^^^^^^^^^^^^ C131
15 |    use testdrive
16 |    use tjson_lexer
17 |    use tomlf_constants, only : tfi, tfr, nl => TOML_NEWLINE
   |

test/unit/lexer.f90:14:1: C131 module 'tftest_lexer' missing default accessibility statement
   |
12 | ! limitations under the License.
13 |
14 | module tftest_lexer
   | ^^^^^^^^^^^^^^^^^^^ C131
15 |    use testdrive
16 |    use tomlf_constants, only : tfi, tfr, nl => TOML_NEWLINE
17 |    use tomlf_datetime, only : toml_datetime, toml_date, toml_time, operator(==), to_string
   |

test/unit/parser.f90:14:1: C131 module 'tftest_parser' missing default accessibility statement
   |
12 | ! limitations under the License.
13 |
14 | module tftest_parser
   | ^^^^^^^^^^^^^^^^^^^^ C131
15 |    use testdrive
16 |    use tomlf_build, only : get_value
17 |    use tomlf_constants, only : nl => TOML_NEWLINE
   |
````
